File: build-logic/convention/src/main/java/org/geogebra/Resources.java
package org.geogebra;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.Reader;
import java.nio.charset.StandardCharsets;

public class Resources {

	public static String getString(String resourcePath) throws IOException {
		InputStream in = Resources.class.getResourceAsStream(resourcePath);

		StringBuilder sb = new StringBuilder();
		try (Reader reader = new BufferedReader(
				new java.io.InputStreamReader(in, StandardCharsets.UTF_8))) {
			int c;
			while ((c = reader.read()) != -1) {
				sb.append((char) c);
			}
		}
		return sb.toString();
	}

}

File: build-logic/convention/src/main/kotlin/org/geogebra/gradle/Resources.kt
package org.geogebra.gradle

import java.io.IOException

class Resources {
    companion object {
        @Throws(IOException::class)
        fun getString(resourcePath: String): String {
            val ins = Resources::class.java.getResourceAsStream(resourcePath)
            return ins!!.bufferedReader().use { it.readText() }
        }
    }
}
